plugins/connection/grpc.py

```diff
@@ -0,0 +1,292 @@
+        be used.
+    ini:
+      - section: defaults
+        key: remote_port
+    env:
+      - name: ANSIBLE_REMOTE_PORT
+    vars:
+      - name: ansible_port
+  network_os:
+    description:
+      - Configures the device platform network operating system. This value is
+        used to load a device specific grpc plugin to communicate with the remote
+        device.
+    vars:
+      - name: ansible_network_os
+  remote_user:
+    description:
+      - The username used to authenticate to the remote device when the gRPC
+        connection is first established.  If the remote_user is not specified,
+        the connection will use the username of the logged in user.
+      - Can be configured from the CLI via the C(--user) or C(-u) options.
+    ini:
+      - section: defaults
+        key: remote_user
+    env:
+      - name: ANSIBLE_REMOTE_USER
+    vars:
+      - name: ansible_user
+  password:
+    description:
+      - Configures the user password used to authenticate to the remote device
+        when first establishing the gRPC connection.
+    vars:
+      - name: ansible_password
+      - name: ansible_ssh_pass
+  private_key_file:
+    description:
+      - The PEM encoded private key file used to authenticate to the
+        remote device when first establishing the grpc connection.
+    ini:
+      - section: grpc_connection
+        key: private_key_file
+    env:
+      - name: ANSIBLE_PRIVATE_KEY_FILE
+    vars:
+      - name: ansible_private_key_file
+  root_certificates_file:
+    description:
+      - The PEM encoded root certificate file used to create a SSL-enabled
+        channel, if the value is None it reads the root certificates from
+        a default location chosen by gRPC at runtime.
+    ini:
+      - section: grpc_connection
+        key: root_certificates_file
+    env:
+      - name: ANSIBLE_ROOT_CERTIFICATES_FILE
+    vars:
+      - name: ansible_root_certificates_file
+  certificate_chain_file:
+    description:
+      - The PEM encoded certificate chain file used to create a SSL-enabled
+        channel. If the value is None, no certificate chain is used.
+    ini:
+      - section: grpc_connection
+        key: certificate_chain_file
+    env:
+      - name: ANSIBLE_CERTIFICATE_CHAIN_FILE
+    vars:
+      - name: ansible_certificate_chain_file
+  ssl_target_name_override:
+    description:
+      - The option overrides SSL target name used for SSL host name checking.
+        The name used for SSL host name checking will be the target parameter
+        (assuming that the secure channel is an SSL channel). If this parameter is
+        specified and the underlying is not an SSL channel, it will just be ignored.
+    ini:
+      - section: grpc_connection
+        key: ssl_target_name_override
+    env:
+      - name: ANSIBLE_GPRC_SSL_TARGET_NAME_OVERRIDE
+    vars:
+      - name: ansible_grpc_ssl_target_name_override
+  persistent_connect_timeout:
```

Comment:
The option can be referred as an extended doc fragment https://github.com/ansible-collections/ansible.netcommon/blob/main/plugins/connection/network_cli.py#L21

Reply:
Done